tests/test_self.py

```diff
@@ -1319,11 +1281,220 @@ def test_regression_recursive_current_dir(self):
                     expected_output="No such file or directory",
                 )
 
+
+class TestCallbackOptions:
+    """Test for all callback options we support."""
+
     @staticmethod
-    def test_long_help(capsys: pytest.CaptureFixture[str]) -> None:
-        """Test the long help flag."""
-        with pytest.raises(SystemExit) as ex:
-            Run(["--long-help"])
-        assert ex.value.code == 0
-        output = capsys.readouterr()
-        assert "Environment variables:" in output.out
+    def test_help_msg() -> None:
+        """Test the --help-msg flag."""
+        process = subprocess.run(
+            [
+                sys.executable,
+                "-m",
+                "pylint",
+                "--help-msg",
+                "W0101",
+            ],
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            check=False,
+        )
+        assert ":unreachable (W0101)" in process.stdout.decode(sys.getdefaultencoding())
+
+        process = subprocess.run(
+            [
+                sys.executable,
+                "-m",
+                "pylint",
+                "--help-msg",
+                "WX101",
+            ],
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            check=False,
+        )
+        assert "No such message id" in process.stdout.decode(sys.getdefaultencoding())
+
+    @staticmethod
+    def test_list_msgs() -> None:
+        """Test the --list-msgs flag."""
+        process = subprocess.run(
+            [
+                sys.executable,
+                "-m",
+                "pylint",
+                "--list-msgs",
+            ],
+            stdout=subprocess.PIPE,
```

Comment:
Not sure about this but worth a try I guess ? 
```suggestion
            stdout=subprocess.PIPE,
            encoding="utf8",
```
https://docs.python.org/3/library/subprocess.html#subprocess.run

> If encoding or errors are specified, or text is true, file objects for stdin, stdout and stderr are opened in text mode using the specified encoding and errors or the [io.TextIOWrapper](https://docs.python.org/3/library/io.html#io.TextIOWrapper) default. The universal_newlines argument is equivalent to text and is provided for backwards compatibility. By default, file objects are opened in binary mode.

Reply:
Still crashes :(

I can't really figure out what the character is we're failing on...

Reply:
Can we print the output ?

Reply:
No since the error is now happening inside `run()`

Reply:
Got access to a windows laptop. Turns out we were using an unsupported character.

Reply:
Which one, the suspense is killing me ? 😄